File: AoC_2020/scripts/script01.R
library(tidyverse)
d1 <- readLines("data/data01") %>% as.numeric()

# star 1
expand_grid(n1 = d1, n2 = d1) %>% 
  filter(n1 + n2 == 2020) %>% 
  summarise(n1 * n2) 
             
# star 2
expand_grid(n1 = d1, n2 = d1, n3 = d1) %>% 
  filter(n1 + n2 + n3 == 2020) %>% 
  summarise(n1 * n2 * n3)

File: AoC_2015/day_01/script01.py
with open('data01.txt') as d:
    data = [el.strip() for el in d.readlines()][0]

floor = 0
for d in data:
    if d == '(':
        floor += 1
    else:
        floor -= 1

print("Star 1:", floor)

floor = 0
cnt = 0
for d in data:
    cnt += 1
    if d == '(':
        floor += 1
    else:
        floor -= 1
    if floor < 0:
        break

print("Star 2:", cnt)